### PR TITLE
[Keyboard] Add steamroller

### PR DESCRIPTION
**Pull request by DeskDaily**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/nightly_boards/steamroller/readme.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# DeskDaily Steamroller
+
+![Steamroller](https://i.imgur.com/x8CmwWD.png)
```
Comment (drashna) [inline keyboards/nightly_boards/steamroller/readme.md:3]:
```suggestion
![Steamroller](https://i.imgur.com/x8CmwWDh.png)
```

File: keyboards/nightly_boards/steamroller/readme.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+A 40% + numpad keyboard
+
+* Keyboard Maintainer: [DeskDaily](https://github.com/DeskDaily)
+* Hardware Supported: Atmega32u4
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/nightly_boards/steamroller/readme.md:10]: Could you add instructions on how to enter the bootloader, to flash new firmware?  This is needed, as a new user may not be aware of how to enter to bootloader to flash new firmware.

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by tzarc** [comment]
Our fault, stalebot kicked in before we got back to it.

**Review by drashna — APPROVED** [review]

File: keyboards/nightly_boards/steamroller/config.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+ */
+
```
Comment (waffle87) [inline keyboards/nightly_boards/steamroller/config.h:35]:
```suggestion
```

File: keyboards/nightly_boards/steamroller/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS  
+    ),
+
+    [3] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS  
+    ),
```
Comment (waffle87) [inline keyboards/nightly_boards/steamroller/keymaps/default/keymap.c:46]:
```suggestion
    )
```

File: keyboards/nightly_boards/steamroller/rules.mk [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
Comment (waffle87) [inline keyboards/nightly_boards/steamroller/rules.mk:6]:
```suggestion
```

File: keyboards/nightly_boards/steamroller/info.json [hunk 6]
```diff
@@ -0,0 +1,83 @@
+{
+    "keyboard_name": "Steamroller", 
+    "manufacturer": "DeskDaily",
+    "url": "", 
+    "maintainer": "DeskDaily", 
+    "usb": {
+    	"vid": "0xD812",
+    	"pid": "0x0023",
+    	"device_version": "0.0.1"
+    },
```
Comment (waffle87) [inline keyboards/nightly_boards/steamroller/info.json:10]:
```suggestion
    },
    "processor": "atmega32u4",
    "bootloader": "atmel-dfu",
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**